### PR TITLE
Add version number in OS settings menu

### PR DESCRIPTION
**Pull request by jedireza**
We're adding the version of the app as a OS level setting in hopes that it helps fix the deep linking to Neeva settings bug we've observed in user testing.

### How was this tested?
- [x] I tested this manually

### Manual test cases
I tested in the simulator for the version number confirmation and I tested the version bumping on the terminal using:

```sh
$ ./Scripts/update-version.sh
```

### Screenshots and screen recordings

image

## Issues addressed
🔒 https://app.clickup.com/t/2xw5aw6

### CODE REVIEW

File: Client/Application/Settings.bundle/Root.plist [hunk 1]
```diff
@@ -2,26 +2,36 @@
...
+			<key>Title</key>
+			<string>Skip Session Restore</string>
+			<key>Type</key>
+			<string>PSToggleSwitchSpecifier</string>
 		</dict>
 	</array>
+	<key>StringsTable</key>
+	<string>Root</string>
 </dict>
 </plist>
```
Comment (jedireza) [inline Client/Application/Settings.bundle/Root.plist:37]: FYI: the noisy diff in this file was from `PlistBuddy` auto formatting. Going forward it should be a one line change when we bump versions.

**Comment by jedireza** [comment]
Currently we conditionally include `Client/Application/Settings.bundle` for debug builds only. We'll need to change that to see this through. But we also want to pull the debug settings out. I'm looking at that next. If you have any ideas let me know.

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6381,7 +6381,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\nfi\n";
+			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\n    # remove the debug settings\n    /usr/libexec/PlistBuddy -c \"Delete :PreferenceSpecifiers:1\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/Settings.bundle/Root.plist\"\n    /usr/libexec/PlistBuddy -c \"Delete :PreferenceSpecifiers:1\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/Settings.bundle/Root.plist\"\nfi\n";
```
Comment (jedireza) [inline Client.xcodeproj/project.pbxproj:6384]: I could use some help testing that this works as expected.

File: Client.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -6381,7 +6381,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\nfi\n";
+			shellScript = "if [ \"${INCLUDE_SETTINGS_BUNDLE}\" = \"YES\" ]\nthen\n    cp -r \"${PROJECT_DIR}/${TARGET_NAME}/Application/Settings.bundle\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\"\n    # remove the debug settings\n    if ![ \"${CONFIGURATION}\" = \"Debug\" ]\n    then\n        /usr/libexec/PlistBuddy -c \"Delete :PreferenceSpecifiers:1\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/Settings.bundle/Root.plist\"\n        /usr/libexec/PlistBuddy -c \"Delete :PreferenceSpecifiers:1\" \"${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app/Settings.bundle/Root.plist\"\n    fi\nfi\n";
```
Comment (jedireza) [inline Client.xcodeproj/project.pbxproj:6384]: I could use some help testing that this works as expected during a release build.

Comment (jedireza) [reply Client.xcodeproj/project.pbxproj:6384]: @darinwf and I spoke about this and we're just going to nuke the debug settings.

File: Client/Application/Settings.bundle/Root.plist [hunk 4]
```diff
@@ -2,26 +2,36 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>StringsTable</key>
-	<string>Root</string>
 	<key>PreferenceSpecifiers</key>
 	<array>
 		<dict>
+			<key>DefaultValue</key>
+			<string>1.41.1</string>
```
Comment (chung-lai) [inline Client/Application/Settings.bundle/Root.plist:9]: If we are nuking the debug setting, can this be `$(BROWSER_MARKETING_VERSION)` so it picks up the value from the `AppConfig.xcconfig`? (If that works, we don't need the plistbuddy command?)

Comment (jedireza) [reply Client/Application/Settings.bundle/Root.plist:9]: From what I've read and understand, this file is going to be copied wholesale as part of the build. If we want dynamic values or conditionals, it needs to be instrumented by us.

Comment (darinwf) [reply Client/Application/Settings.bundle/Root.plist:9]: Yeah, that's what I was wondering too

Comment (darinwf) [reply Client/Application/Settings.bundle/Root.plist:9]: Guess not :-)

Simulator Screen Shot - iPhone 13 - 2022-05-16 at 21 59 18

**Review by darinwf — APPROVED** [review]

**Pull request merged**